source/fuzz/transformation_access_chain.cpp

```diff
@@ -136,24 +136,50 @@ bool TransformationAccessChain::IsApplicable(
       }
 
       // Get two new ids to use and update the amount used.
-      protobufs::UInt32Pair pair =
+      protobufs::UInt32Pair fresh_ids =
           message_.fresh_ids_for_clamping()[id_pairs_used++];
 
       // Check that the ids are actually fresh and not already used by this
       // transformation.
       if (!CheckIdIsFreshAndNotUsedByThisTransformation(
-              pair.first(), ir_context, &fresh_ids_used) ||
+              fresh_ids.first(), ir_context, &fresh_ids_used) ||
           !CheckIdIsFreshAndNotUsedByThisTransformation(
-              pair.second(), ir_context, &fresh_ids_used)) {
+              fresh_ids.second(), ir_context, &fresh_ids_used)) {
         return false;
       }
 
-      if (!CreateAndGetClampedIndexId(ir_context, index_id, subobject_type_id,
-                                      false, {pair.first(), pair.second()})
-               .first) {
+      if (!ValidIndexToComposite(ir_context, index_id, subobject_type_id)) {
         return false;
       }
 
+      // Valid fresh ids need to have been given
+      if (fresh_ids.first() == 0 || fresh_ids.second() == 0) {
+        return false;
+      }
```

Comment:
I suggest moving this up to line 141 - right after you get the fresh ids, and change the comment to "Valid ids need to have been given".

Reply:
Done